### PR TITLE
fix lnurl channel

### PR DESCRIPTION
**Pull request by pavanjoshi914**
show proper error messages

### Describe the changes you have made in this PR

fix lnurl callback
proper error messages while doing requests

### Type of change

(Remove other not matching type)

- `fix`: Bug fix (non-breaking change which fixes an issue)

### Checklist

- [X] Self-review of changed code
- [X] Manual testing
- [X] Added automated tests where applicable
- [X] Update Docs & Guides
- For UI-related changes
- [X] Darkmode
- [X] Responsive layout

### CODE REVIEW

File: src/app/screens/LNURLChannel/index.tsx [hunk 1]
```diff
@@ -73,8 +76,12 @@ function LNURLChannel() {
         msg.reply(callbackResponse?.data);
       }
     } catch (e) {
-      console.error(e);
-      if (e instanceof Error) toast.error(`Error: ${e.message}`);
+      if (axios.isAxiosError(e)) {
```
Comment (reneaaron) [inline src/app/screens/LNURLChannel/index.tsx:79]: Why only if it's an axios error? if e.g. api.getInfo() fails there is no error shown?

File: src/app/screens/LNURLChannel/index.tsx [hunk 2]
```diff
@@ -115,6 +122,22 @@ function LNURLChannel() {
...
+                checked={isPrivateChannel}
+                onChange={(event) => {
+                  setIsPrivateChannel(event.target.checked);
+                }}
+              />
+              <label
+                htmlFor="is_private_channel"
+                className="cursor-pointer ml-2 block text-sm text-gray-900 font-medium dark:text-white"
+              >
+                Open a private channel?
```
Comment (reneaaron) [inline src/app/screens/LNURLChannel/index.tsx:138]: Add to translations

File: src/app/screens/LNURLChannel/index.tsx [hunk 3]
```diff
@@ -115,6 +122,22 @@ function LNURLChannel() {
               heading={`${t("content_message.heading")}:`}
               content={uri}
             />
+            <div className="flex">
+              <Checkbox
+                id="is_private_channel"
+                name="private channel checkbox"
```
Comment (reneaaron) [inline src/app/screens/LNURLChannel/index.tsx:128]: Can we use a proper name here? (it's allowed and works, but to me this feels strange)

Comment (reneaaron) [reply src/app/screens/LNURLChannel/index.tsx:128]:
```suggestion
                name="is_private_channel_checkbox"
```

**Review by reneaaron — CHANGES_REQUESTED** [review]
Nice! :rocket:

**Comment by pavanjoshi914** [comment]
sry i just quickly coded it. need to improve the pr a bit. thanks for the review addressing it

File: src/app/screens/LNURLChannel/index.tsx [hunk 4]
```diff
@@ -74,7 +77,13 @@ function LNURLChannel() {
       }
     } catch (e) {
       console.error(e);
-      if (e instanceof Error) toast.error(`Error: ${e.message}`);
+      if (axios.isAxiosError(e)) {
+        const error =
+          (e.response?.data as { reason?: string })?.reason || e.message;
+        toast.error(`Error: ${error}`);
+      } else if (e instanceof Error) {
+        toast.error(`Error: ${e.message}`);
```
Comment (reneaaron) [inline src/app/screens/LNURLChannel/index.tsx:85]: If we add this here we should be able to remove https://github.com/getAlby/lightning-browser-extension/pull/3046/files#diff-a082864a4adc3604b70d8b843b52fc353f4debd06482a72388a209bb6701e1f2R47-R49, right?

**Review by reneaaron — APPROVED** [review]
One little thing to change. 

Otherwise utACK

File: src/app/screens/LNURLChannel/index.tsx (hunk 4 above)
Comment (pavanjoshi914) [reply src/app/screens/LNURLChannel/index.tsx:82]: yup true! that's not needed it will eventually be catched and shown in toast. thanks!

**Comment by pavanjoshi914** [comment]
will merge this. i think functionality is complete now. if there are further improvements we need to do. i can do a follow up pr

**Pull request merged**